Adding Parameter Sweep for Two Hidden Layer Model / Reorganizing nbconverted Scripts

gwaygenomics opened the pull request:
A lot of intermediate files being added in the PR in commit 9997c6d. These files are the result of a parameter sweep of a two hidden layer model (5000 -> 100 -> 100 -> 100 -> 5000). Commit 2b61e96 adds the compiled file with git LFS.

The main file changes include adding options to the parameter sweep script (cd77571) and adding a script that actually trains the models (7abeb7b). This script is very similar to the tybalt notebook.

jaclyn-taroni commented on scripts/vae_pancancer.py:
```diff
@@ -0,0 +1,222 @@
+first_layer = int(args.first_layer)
+output_filename = args.output_filename
+
+original_dim = 5000
```
Might change if no. genes based on MAD filter changes?

gwaygenomics replied:
:+1:

jaclyn-taroni commented on scripts/vae_pancancer.py:
```diff
@@ -0,0 +1,222 @@
+# ~~~~~~~~~~~~~~~~~~~~~~
+# ENCODER
+# ~~~~~~~~~~~~~~~~~~~~~~
+# Input layer is compressed into a mean and log variance vector of size
```
Does this need to be updated to take into account the two hidden layers? Hard to tell

jaclyn-taroni commented on scripts/vae_pancancer.py:
```diff
@@ -0,0 +1,222 @@
+# ~~~~~~~~~~~~~~~~~~~~~~
+# DECODER
+# ~~~~~~~~~~~~~~~~~~~~~~
+# The decoding layer is much simpler with a single layer glorot uniform
```
Same comment as above

danich1 commented:
I agree with @jaclyn-taroni comments. Just curious as to why a completely new script instead of modifying `tybalt_vae.py`?

jaclyn-taroni commented:
I have the same question as @danich1, that's what I was trying to get at. I thought it might be for notebook reasons?

gwaygenomics commented:
> Just curious as to why a completely new script instead of modifying tybalt_vae.py?

> I thought it might be for notebook reasons?

Precisely. I think it would be best to keep the nbconverted scripts separate from scripts run consistently. If they are not separate, when a notebook changes, it would get messy quick.

Do you think the repo would be better organized with a `scripts/nbcovert/` folder storing these scripts instead?

danich1 commented:
I concur on separating out the nbconverted scripts. Better to separate them off to avoid confusion about which is suppose to be a converted script and what is suppose to be a run-consistent script.

gwaygenomics commented:
thanks @danich1 and @jaclyn-taroni - I believe b895ece addresses your concerns. I also believe that some portion of concern is regarding redundancy and I definitely agree. 

Portions of this code base are getting entirely too redundant, and this realization prompts refactorization (see #74 for more details).

gwaygenomics commented:
> I concur on separating out the nbconverted scripts. Better to separate them off to avoid confusion about which is suppose to be a converted script and what is suppose to be a run-consistent script.

Sounds good. in the next commit, inside this PR (I will also rename the PR). I will reorganize nbconverted scripts.

gwaygenomics commented:
thanks @danich1 and @jaclyn-taroni - ready for review again

jaclyn-taroni commented:
@gwaygenomics - Does your output change when setting `kernel_initializer='glorot_uniform'` in your decoder in [`b895ece`](https://github.com/greenelab/tybalt/pull/71/commits/b895ecea5ea7988e0e856a41a25341d8dbdc6df5)? If so, should probably update those files as well here

gwaygenomics commented:
> Does your output change when setting kernel_initializer='glorot_uniform' in your decoder in b895ece?

Good catch. It does not change since [`glorot_uniform` is the default](https://keras.io/layers/core/). I add it here for consistency and to remove those concerns.
